app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -269,11 +271,37 @@ def build_metric_id(type, resource, metric_id)
     end
 
     def self.update_alert(*args)
-      _log.debug("Updating an alert on Hawkular provider: #{args}")
+      operation = args[0][:operation]
+      alert = args[0][:alert]
+      miq_alert = {
+        :id          => alert[:id],
+        :enabled     => alert[:enabled],
+        :description => alert[:description],
+        :conditions  => alert[:expression]
+      }
+      MiddlewareManager.find_each { |m| m.alert_manager.process_alert(operation, miq_alert) }
     end
 
     def self.update_alert_profile(*args)
-      _log.debug("Updating an alert profile on Hawkular provider: #{args}")
+      alert_profile_arg = args[0]
+      miq_alert_profile = {
+        :id                  => alert_profile_arg[:profile_id],
+        :old_alerts_ids      => alert_profile_arg[:old_alerts],
+        :new_alerts_ids      => alert_profile_arg[:new_alerts],
+        :old_assignments_ids => process_old_assignments_ids(alert_profile_arg[:old_assignments]),
+        :new_assignments_ids => process_new_assignments_ids(alert_profile_arg[:new_assignments])
+      }
+      MiddlewareManager.find_each do |m|
+        m.alert_profile_manager.process_alert_profile(alert_profile_arg[:operation], miq_alert_profile)
```

Comment:
I am probably misunderstanding something: you loop over all middleware managers (=MW providers) and then do something on the `alert_profile_manager`, which is instantiated in line 304 if it does not yet exist.
You pass `self`, which is the current MW-manager. But shouldn't this be an `AlertProfileManager` for the respective MW-manager `m` ? (Same for `AlertManager`)